src/main/java/core/basesyntax/service/impl/QuantityServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.QuantityService;
+import java.util.Map;
+
+public class QuantityServiceImpl implements QuantityService {
+    private static final Map<Fruit, Integer> STORAGE = Storage.fruits;
+    private Integer variable;
+
+    @Override
+    public void add(Fruit fruit, Integer quantity) {
+        variable = STORAGE.containsKey(fruit)
+                ? STORAGE.replace(fruit, STORAGE.get(fruit) + quantity) :
+                STORAGE.put(fruit, quantity);
+    }
+
+    @Override
+    public void subtract(Fruit fruit, Integer quantity) {
+        variable = STORAGE.containsKey(fruit)
+                ? STORAGE.replace(fruit, STORAGE.get(fruit) - quantity) :
+                STORAGE.put(fruit, -quantity);
```

Comment:
do you think we can buy the fruit when it's no fruit present? Or we should throw exception or something?